src/main/java/com/bobocode/hoverla/bring/context/ApplicationContextImpl.java

```diff
@@ -1,60 +1,152 @@
 package com.bobocode.hoverla.bring.context;
 
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
+import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.Objects;
+
 
 /**
- * Example of ApplicationContext implementation, can be removed or reimplemented when actual development starts
+ * Implementation of ApplicationContext
  */
+@Slf4j
 public class ApplicationContextImpl implements ApplicationContext {
+    private Table<String, Class<?>, BeanDefinition> context = HashBasedTable.create();
+    private final String BEAN_TYPE_MUST_BE_NOT_NULL_MESSAGE = "The argument [beanType] must be not null";
```

Comment:
No need to extract these messages as constants. If they are used in multiple places, they should be fetched from somewhere - property file etc.  Use them directly in exception constructor for now.